### ISSUE TITLE
openssl rehash fails on non existing certificates

### ISSUE THREAD

**Issue by sebastianas** [openssl/openssl#6475]
I backported #6083 to current Debian unstable due to [#895482](https://bugs.debian.org/895482).
There still seems to be on difference in the behavior of "openssl rehash" vs "c_rehash":

> /etc/ssl/certs# ln -sv /notting Swisscom_Root_CA_1.pem
> 'Swisscom_Root_CA_1.pem' -> '/notting'
> /etc/ssl/certs# ls -lh Swisscom_Root_CA_1.pem
> lrwxrwxrwx 1 root root 8 Jun 12 23:29 Swisscom_Root_CA_1.pem -> /notting
> /etc/ssl/certs# openssl rehash
> rehash: error: skipping Swisscom_Root_CA_1.pem, cannot open file
> rehash: warning: skipping ca-certificates.crt,it does not contain exactly one certificate or CRL
> /etc/ssl/certs# echo $?
> 1
> /etc/ssl/certs# c_rehash
> Doing /usr/lib/ssl/certs
> WARNING: Swisscom_Root_CA_1.pem does not contain a certificate or CRL: skipping
> WARNING: Skipping duplicate certificate ca-certificates.crt
> WARNING: Skipping duplicate certificate ca-certificates.crt
> root@bazinga:/etc/ssl/certs# echo $?
> 0

The problematic is:

```
 static int do_file(const char *filename, const char *fullpath, enum Hash h)
 {
…
    /* Does it have X.509 data in it? */
     if ((b = BIO_new_file(fullpath, "r")) == NULL) {
         BIO_printf(bio_err, "%s: error: skipping %s, cannot open file\n",
                    opt_getprog(), filename);
         errs++;
         goto end;
     }
 
```
May I drop that "errs++" in order to align with "c_rehash" behaviour?

**Comment by levitte** [comment]
It's true that `openssl rehash` is stricter than `c_rehash` in some cases (`c_rehash` doesn't check for errors at all when opening files)....  @richsalz, do you have an opinion on this?

**Comment by richsalz** [comment]
no opinion except that i'd like to see the perl script go away some time.

**Comment by otavio** [comment]
This was addressed on 1.1.1; could this to be closed?

**Comment by levitte** [comment]
Sure
